lisa/analysis/rta.py

```diff
@@ -0,0 +1,339 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2018, Arm Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from numpy import unique
+
+from lisa.utils import memoized
+from lisa.analysis.base import AnalysisHelpers
+from lisa.analysis.base import TraceAnalysisBase
+from lisa.trace import requires_events
+
+class Events(TraceAnalysisBase):
+    """
+    Support for RTA events analysis.
+
+    :param trace: input Trace object
+    :type trace: :class:`trace.Trace`
+    """
+
+    name = 'rta'
+
+    def __init__(self, trace):
+        """
+        Load peformance data of an rt-app workload
+        """
+        logger = self.get_logger()
+
+        self.trace = trace
+
+        if 'rtapp_task' not in self.trace.base_trace.available_events:
```

Comment:
s/self.trace.base_trace/self.trace/

`base_trace` is an attribute of TraceView only, analysis need to work with regular traces as well.